common/src/main/java/io/homeassistant/companion/android/common/sensors/LastUpdateManager.kt

```diff
@@ -98,6 +98,9 @@ class LastUpdateManager : SensorManager {
                 // add the new Intent:
                 sensorDao.add(SensorSetting(lastUpdate.id, newIntentSettingName, intentAction, SensorSettingType.STRING))
             }
+        } else {
+            // add the toggle if it was not already added.
+            sensorDao.add(SensorSetting(lastUpdate.id, SETTING_ADD_NEW_INTENT, "false", SensorSettingType.TOGGLE))
```

Comment:
The execution will reach the `else` branch when ...
a. there's no toggle yet
b. **there's already a toggle, with `false` value**

So I think this would cause a DB write operation on every sensor update.

Maybe we could do something like this to prevent this:

```kotlin
val addNewIntentToggle = allSettings.firstOrNull { it.name == SETTING_ADD_NEW_INTENT }
if (addNewIntentToggle == null) {
    // add the toggle if it was not already added.
    sensorDao.add(SensorSetting(lastUpdate.id, SETTING_ADD_NEW_INTENT, "false", SensorSettingType.TOGGLE))
} else {
    if (addNewIntentToggle.value == "true") {
        val newIntentSettingOrdinal = intentSettings.size + 1
        val newIntentSettingName = "$INTENT_SETTING_PREFIX$newIntentSettingOrdinal:"
        if (allSettings.none { it.name == newIntentSettingName }) {
            // turn off the toggle:
            sensorDao.add(SensorSetting(lastUpdate.id, SETTING_ADD_NEW_INTENT, "false", SensorSettingType.TOGGLE))
            // add the new Intent:
            sensorDao.add(SensorSetting(lastUpdate.id, newIntentSettingName, intentAction, SensorSettingType.STRING))
        }
    }
}
```

Reply:
Good call, the old code always wrote to the DB here.  Just tested this code and its working as expected on a new install. Will update  now.

Reply:
Sorry, I didn't state it explicitly that I edited my code suggestion. It's up to you if you like the new version more or the already committed version.

Reply:
> Sorry, I didn't notify you that I edited my code suggestion. It's up to you if you like the new version more or the already committed version.

LOL I just saw this, lets leave it as is for now to get the fix out there. We can look into improving this later I think :)